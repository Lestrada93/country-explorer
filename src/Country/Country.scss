@import "../assets/colors";
.country {
    width: 100%;
    cursor: pointer;
    padding: 0.5rem 1rem;
    color: $text;
    text-decoration: none;
    background-color: #fff;
    border: 1px solid $light-border;
}