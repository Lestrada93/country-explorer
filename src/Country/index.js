import React from "react";
import PropTypes from "prop-types";
import { withTranslation } from "react-i18next";
import "./Country.scss";

/**
 * Country
 * @description It shows the country information
 * @param {*} param0 
 * @returns 
 */
function Country({ t, data, seeDetails }) {
    return <button
        className="country"
        onClick={() => seeDetails(data.id)}>
        {data?.flag}
        <span className="country__name">{data.name}</span>
        {
            data?.counter &&
            <span>{t("VIEWED")}: {data?.counter}</span>
        }
    </button>;
}

Country.propTypes = {
    t: PropTypes.func,
    data: PropTypes.object.isRequired,
    seeDetails: PropTypes.func.isRequired,
};

export default withTranslation()(Country);