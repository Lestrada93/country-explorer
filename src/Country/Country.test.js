import React from 'react';
import { create } from 'react-test-renderer';
import Country from '.';

describe('Country component', () => {
    test('Snapshot', () => {
        const country = create(<Country data={{ name: "test" }} seeDetails={() => { }} />);
        expect(country.toJSON()).toMatchSnapshot();
    });
});
