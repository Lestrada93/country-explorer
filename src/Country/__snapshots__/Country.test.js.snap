// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country component Snapshot 1`] = `
<button
  className="country"
  onClick={[Function]}
>
  <span
    className="country__name"
  >
    test
  </span>
</button>
`;
